src/main/scala/gestalt/Symbols.scala

```diff
@@ -15,5 +15,8 @@ trait Symbols { this: Toolbox =>
 
     /** type of a member with respect to a prefix */
     def asSeenFrom(mem: Symbol, prefix: Type): Type
+
+    /** subst symbols in tree */
+    def subst(tree: tpd.Tree)(from: List[Symbol], to: List[Symbol]): tpd.Tree
```

Comment:
I wonder whether there is another way to implement monadless-like macros without going through the manual typechecking route. Even in reflect, which is pretty low-level, APIs like `currentOwner`, `newSymbol`, etc belong to `c.internal`.